File: repos_estudiantes_ALVAREZ_segunda_entrega/GUTIERREZ_GARCIA_CARLA_ALEJANDRA/Herencia/Ejercicio1/Vehiculo.java
package HERENCIA.Ejercicio1;

public class Vehiculo {
    protected String marca, modelo;
    protected int año;
    protected float precio_base;

    public Vehiculo(String m, String mo, int a, float pb) {
        this.marca = m;
        this.modelo = mo;
        this.año = a;
        this.precio_base = pb;

    }

    public String getMarca() {
        return marca;
    }

    public String getModelo() {
        return modelo;
    }

    public int getAño() {
        return año;
    }

    public float getPrecio_base() {
        return precio_base;
    }

    public void setMarca(String m) {
        this.marca = m;
    }

    public void setModelo(String mo) {
        this.modelo = mo;
    }

    public void setAño(int a) {
        this.año = a;
    }

    public void setPrecio_base(float pb) {
        this.precio_base = pb;
    }

    public void mostrar_info() {
        System.out.println("Marca: " + this.marca + ", Modelo: " + this.modelo + ", Año: " + this.año
                + ", Precio Base: " + this.precio_base);
    }

    public void mostrar_actual(int x) {

    }

}

File: repos_estudiantes_ALVAREZ_segunda_entrega/GUTIERREZ_GARCIA_CARLA_ALEJANDRA/Herencia/Ejercicio1/Vehiculo.py
class Vehiculo:
    def __init__(self, m, mo, a, pb):
        self.marca = marca
        self.modelo = modelo
        self.año = año
        self.precio_base = precio_base

    def mostrar_info(self):
        print(f"Marca: {self.marca}, Modelo: {self.modelo}, Año: {self.año}, Precio Base: {self.precio_base}")

class Coche(Vehiculo):
    def __init__(self, marca, modelo, año, precio_base, num_puertas, tipo_combustible):
        super().__init__(marca, modelo, año, precio_base)
        self.num_puertas = num_puertas
        self.tipo_combustible = tipo_combustible

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Num Puertas: {self.num_puertas}, Tipo Combustible: {self.tipo_combustible}")

class Moto(Vehiculo):
    def __init__(self, marca, modelo, año, precio_base, tipo_motor, cilindrada):
        super().__init__(marca, modelo, año, precio_base)
        self.tipo_motor = tipo_motor
        self.cilindrada = cilindrada

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Tipo Motor: {self.tipo_motor}, Cilindrada: {self.cilindrada}")

# a) Instancias de Coche y Moto
moto1 = Moto("Yamaha", "MT-07", 2023, 7500.0, "Deportiva", 689)
moto2 = Moto("Honda", "CB500X", 2025, 8000.0, "Adventure", 471)
coche1 = Coche("Toyota", "Corolla", 2022, 20000.0, 4, "Gasolina")
coche2 = Coche("Ford", "Explorer", 2025, 35000.0, 5, "Diesel")

vehiculos = [moto1, moto2, coche1, coche2]

# b) Mostrar información de todos los vehículos
print("Información de todos los vehículos:")
for v in vehiculos:
    v.mostrar_info()
    print()

# c) Mostrar coches con más de 4 puertas
print("Coches con más de 4 puertas:")
for v in vehiculos:
    if isinstance(v, Coche) and v.num_puertas > 4:
        v.mostrar_info()
        print()

# d) Mostrar vehículos actuales (gestión 2025)
print("Vehículos actuales (gestión 2025):")
for v in vehiculos:
    if v.año == 2025:
        v.mostrar_info()
        print()